pkg/ee/default-policy-controller/controller.go

```diff
@@ -0,0 +1,399 @@
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	ControllerName = "kkp-default-policy-controller"
+)
+
+type Reconciler struct {
+	ctrlruntimeclient.Client
+
+	workerName   string
+	recorder     record.EventRecorder
+	seedGetter   provider.SeedGetter
+	configGetter provider.KubermaticConfigurationGetter
+	log          *zap.SugaredLogger
+	versions     kubermatic.Versions
+}
+
+func Add(ctx context.Context, mgr manager.Manager, numWorkers int, workerName string, seedGetter provider.SeedGetter, kubermaticConfigurationGetter provider.KubermaticConfigurationGetter, log *zap.SugaredLogger, versions kubermatic.Versions) error {
+	reconciler := &Reconciler{
+		Client: mgr.GetClient(),
+
+		workerName:   workerName,
+		recorder:     mgr.GetEventRecorderFor(ControllerName),
+		seedGetter:   seedGetter,
+		configGetter: kubermaticConfigurationGetter,
+		log:          log,
+		versions:     versions,
+	}
+
+	_, err := builder.ControllerManagedBy(mgr).
+		Named(ControllerName).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: numWorkers,
+		}).
+		// Watch for clusters
+		For(&kubermaticv1.Cluster{}).
+		// Watch changes for PolicyTemplates that have been enforced.
+		Watches(&kubermaticv1.PolicyTemplate{}, enqueueClusters(reconciler, log), builder.WithPredicates(withEventFilter())).
+		Build(reconciler)
+
+	return err
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	cluster := &kubermaticv1.Cluster{}
+	if err := r.Get(ctx, request.NamespacedName, cluster); err != nil {
+		if apierrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	if cluster.DeletionTimestamp != nil {
+		// Cluster deletion in progress, skipping reconciliation
+		r.log.Debugw("Cluster deletion in progress, skipping reconciliation", "cluster", cluster.Name)
+		return reconcile.Result{}, nil
+	}
+
+	result, err := util.ClusterReconcileWrapper(
+		ctx,
+		r,
+		r.workerName,
+		cluster,
+		r.versions,
+		kubermaticv1.ClusterConditionDefaultPolicyControllerReconcilingSuccess,
+		func() (*reconcile.Result, error) {
+			return r.reconcile(ctx, cluster)
+		},
+	)
+
+	if result == nil || err != nil {
+		result = &reconcile.Result{}
+	}
+
+	if err != nil {
+		r.recorder.Event(cluster, corev1.EventTypeWarning, "ReconcilingError", err.Error())
+	}
+
+	return *result, err
+}
+
+func (r *Reconciler) reconcile(ctx context.Context, cluster *kubermaticv1.Cluster) (*reconcile.Result, error) {
+	// Ensure that the cluster is healthy first, this is important for the default policy controller, and for all policy controllers.
+	if !cluster.Status.ExtendedHealth.AllHealthy() {
+		r.log.Debugw("Cluster is not healthy, skipping reconciliation", "cluster", cluster.Name)
+		return &reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	// Determine whether to ignore default policies
+	//nolint:staticcheck
```

Comment:
```
createdCondition := kubermaticv1.ClusterConditionDefaultPolicyBindingsControllerCreatedSuccessfully
ignoreDefaultPolicies := cluster.Status.HasConditionValue(createdCondition, corev1.ConditionTrue)
```

To silence the linter because (I am only guessing here) the one-line variant would be too long, but then also using a pretty long if-statement (simply because that constant is so unwieldy)... sneaky sneaky. But we can do better.

Bonus points: You can re-use `createdCondition` further down to shorten even more lines :)